coalib/settings/FunctionMetadata.py

```diff
@@ -161,6 +161,21 @@ def from_function(cls, func, omit=frozenset()):
                    optional_params=optional_params,
                    omit=omit)
 
+    def filter_parameters(self, dct):
+        """
+        Filters the given dict for keys that are declared as parameters inside
+        this metadata (either optional or non-optional).
+
+        :param dct:
+            The dict to filter.
+        :return:
+            A new dict containing the filtered items.
+        """
+        return {key: dct[key]
+                for key in self.non_optional_params.keys() |
+                self.optional_params.keys()
```

Comment:
I think this self should be aligned with the one above, maybe make brackets around them? They belong together. Romantic.